### PR TITLE
travis: update go version

### PR DESCRIPTION
**Pull request by dvrkps**
1.x match latest stable go version

### CODE REVIEW

**Comment by divan** [comment]
@dvrkps thanks, I didn't know about that `1.x` wildcard. Although there is usually no risks in upgrading Go version, I think we prefer "upgrade Go version" process to be manual and explicit action.

**Comment by dvrkps** [comment]
@divan 1.9.x match latest 1.9 subversion(1.9.2 today)

**Review by tiabc — APPROVED** [review]
Looks good. However, it's not the only place where we build go projects, we'll also need to change it in:
1. https://github.com/status-im/faucet/blob/39999d545a9185b9ab1248f635cea882c9e84692/Dockerfile#L1
2. https://github.com/status-im/status-cluster/blob/18db330d87c5c19300d454f345714d486529eccb/roles/golang/defaults/main.yml#L2
3. https://github.com/status-im/status-cluster/blob/18db330d87c5c19300d454f345714d486529eccb/group_vars/jenkins-nodes.yml#L2

**Review by adambabik — APPROVED** [review]

**Comment by adambabik** [comment]
@tiabc these are private repos and I believe they should have a separate CI integration. Another thing is whether we should run tests on both 1.8.x and 1.9.x Go releases here...

**Comment by dvrkps** [comment]
@tiabc @adambabik @divan i added 1.8.x also.

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,8 @@ notifications:
   email: false
 language: go
 go:
-- 1.9
+- 1.8.x
```
Comment (adambabik) [inline .travis.yml]: As we use Build Stages it's more tricky. You can check [this issue](https://github.com/travis-ci/beta-features/issues/28) for possible solutions. We would like to run `Lint` and `Test unit and integration` on both. The rest of stages can execute on `1.9.x` only.

**Review by adambabik — CHANGES_REQUESTED** [review]

**Review by tiabc — CHANGES_REQUESTED** [review]
@adambabik @dvrkps let's just run everything on a single Go version? I can't see why we may need to run anything on 1.8 and 1.9

**Comment by dvrkps** [comment]
@tiabc @adambabik single version as 1.x or 1.9.x ? 
1.x match latest stable go version

**Comment by adambabik** [comment]
@dvrkps I think `1.9.x` was good. We need more control over upgrading Go version.

**Comment by tiabc** [comment]
1.9.x is fine. We don't need to uncontrollably update the minor version while the patch is fine.

**Comment by dvrkps** [comment]
@tiabc @adambabik done. i remove 1.8.x.

**Review by adambabik — APPROVED** [review]
Thanks!

**Review by tiabc — APPROVED** [review]
Such a long discussion and such a small changelist :)

**Pull request merged**